dlt/pipeline/pipeline.py

```diff
@@ -1707,3 +1716,9 @@ def _save_state(self, state: TPipelineState) -> None:
     def __getstate__(self) -> Any:
         # pickle only the SupportsPipeline protocol fields
         return {"pipeline_name": self.pipeline_name}
+
+    def _dataset(self, dataset_type: Literal["dbapi", "ibis"] = "dbapi") -> SupportsReadableDataset:
```

Comment:
my idea to pass external connection to dataset would be to accept `SqlClientBase` as `dataset_type` in that case we'd create `ReadableDBAPIDataset` with it,

however we can do it directly by getting sql_client form the pipeline already.

Reply:
ok, let's do this in the next iteration, then you can explain what you want to do exactly